src/Tasks/AssemblyDependency/ResolveAssemblyReference.cs

```diff
@@ -1120,9 +1121,11 @@ List<Exception> generalResolutionExceptions
                         if (assemblyFoldersEx != null && _showAssemblyFoldersExLocations.TryGetValue(r.SearchPath, out messageImportance))
                         {
                             Log.LogMessageFromResources(messageImportance, "ResolveAssemblyReference.AssemblyFoldersExSearchLocations", r.SearchPath);
+                            var usedFolders = new HashSet<string>();
                             foreach (AssemblyFoldersExInfo info in assemblyFoldersEx)
                             {
-                                Log.LogMessageFromResources(messageImportance, "ResolveAssemblyReference.EightSpaceIndent", info.DirectoryPath);
+                                if (usedFolders.Add(info.DirectoryPath))
```

Comment:
nit: I know that the [runtime coding style](https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/coding-style.md) allows omitting braces in single-statement `if`'s but I believe that MSBuild follows the convention of always using them.

Reply:
Fixed, by using the suggested helper method.